src/Acuminator/Acuminator.Analyzers/FixProviders/DAC/DacPropertyAttributes/DacExtensionDefaultAttributeFix.cs

```diff
@@ -0,0 +1,138 @@
+		private const string _PersistingCheck = nameof(PXDefaultAttribute.PersistingCheck);
+		private const string _PersistingCheckNothing = nameof(PXPersistingCheck.Nothing);
+
+
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1030_DefaultAttibuteToExisitingRecords.Id);
+
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1030_DefaultAttibuteToExisitingRecords.Id);
+
+			if (diagnostic == null)
+				return;
+
+			if (diagnostic.IsBoundField())
+			{
+				string codeActionNameBound = nameof(Resources.PX1030FixBound).GetLocalized().ToString();
+				CodeAction codeActionBound =
+					CodeAction.Create(codeActionNameBound,
+							cToken => ReplaceIncorrectDefaultAttribute(context.Document, context.Span, diagnostic.IsBoundField(), cToken),
+							equivalenceKey: codeActionNameBound);
+
+				context.RegisterCodeFix(codeActionBound, context.Diagnostics);
+			}
+			else
+			{
+				string codeActionNameUnbound = nameof(Resources.PX1030FixUnbound).GetLocalized().ToString();
+				CodeAction codeActionUnbound =
+					CodeAction.Create(codeActionNameUnbound,
+										cToken => ReplaceIncorrectDefaultAttribute(context.Document, context.Span, diagnostic.IsBoundField(), cToken),
+										equivalenceKey: codeActionNameUnbound);
+
+				context.RegisterCodeFix(codeActionUnbound, context.Diagnostics);
+			}
+		}
+
+		private async Task<Document> ReplaceIncorrectDefaultAttribute(Document document, TextSpan span, bool isBoundField,CancellationToken cancellationToken)
+		{
+			SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+			 if (!(root?.FindNode(span) is AttributeSyntax attributeNode))
+				return document;
+
+			if (!(attributeNode.Parent is AttributeListSyntax attributeList))
+				return document;
+
+			cancellationToken.ThrowIfCancellationRequested();
+
+			SyntaxGenerator generator = SyntaxGenerator.GetGenerator(document);
+
+			var memberAccessExpression = generator.MemberAccessExpression(generator.IdentifierName(_PXPersistingCheck),
+																		  generator.IdentifierName(_PersistingCheckNothing));
+			var persistingAttributeArgument = generator.AttributeArgument(_PersistingCheck, 
+																		  memberAccessExpression) as AttributeArgumentSyntax;
+			var pxUnboundDefaultAttribute = generator.Attribute(_PXUnboundDefaultAttributeName) as AttributeListSyntax;
+			
+			SyntaxNode modifiedRoot;
+
+			if (isBoundField)
+			{
+				if (attributeNode.ArgumentList != null)
+				{
+					var persistingCheckArguments = from arguments in attributeNode.ArgumentList.Arguments
+													where arguments.GetText().ToString().Contains(_PersistingCheck)
+													select arguments;
```

Comment:
@dropsonic I'm not sure that the plain text check instead of syntax check should be used here. What do you think?

Reply:
@saloponov it there a reason for plain text check?
@SENya1990 I think that all checks should be performed on a syntax tree